lib/loglevel.js

```diff
@@ -189,68 +195,108 @@
       self.methodFactory = factory || defaultMethodFactory;
 
       self.getLevel = function () {
-          if (userLevel != null) {
-            return userLevel;
-          } else if (defaultLevel != null) {
-            return defaultLevel;
-          } else {
-            return inheritedLevel;
+          if (userLevel !== null) {
+              return userLevel;
+          }
+          if (defaultLevel !== null) {
+              return defaultLevel;
           }
+          if (inheritedLevel === null) {
+              inheritedLevel = self.getInheritedLevel();
+          }
+          return inheritedLevel;
       };
 
-      self.setLevel = function (level, persist) {
-          userLevel = normalizeLevel(level);
-          if (persist !== false) {  // defaults to true
-              persistLevelIfPossible(userLevel);
+      self.getInheritedLevel = function() {
+          if( defaultLogger && self.parentKey ) {
+              return defaultLogger.getLogger(self.parentKey).getLevel();
           }
-
-          // NOTE: in v2, this should call rebuild(), which updates children.
-          return replaceLoggingMethods.call(self);
+          if( !defaultLogger || defaultLogger===self ) {
+              return normalizeLevel("WARN");
+          }
+          return normalizeLevel(
+              defaultLogger ? defaultLogger.getLevel() : "WARN" );
       };
 
-      self.setDefaultLevel = function (level) {
-          defaultLevel = normalizeLevel(level);
-          if (!getPersistedLevel()) {
-              self.setLevel(level, false);
+      /** The parentKey is the name of the parent this inherits from */
+      self.parentKey = undefined;
+
+      self._clearInheritedLevel = function () {
+          inheritedLevel = null;
+          var loggerNames = Object.keys(_loggersByName);
+          for(var i=0; i<loggerNames.length; i++) {
+              var loggerName = loggerNames[i];
+              var childLogger = _loggersByName[loggerName];
+              if( childLogger.parentKey===name ) {
+                  _loggersByName[loggerName]._clearInheritedLevel();
+              }
           }
       };
 
-      self.resetLevel = function () {
-          userLevel = null;
-          clearPersistedLevel();
-          replaceLoggingMethods.call(self);
+      self.getLogger = function(name) {
+          var childCategories = Array.from(arguments);
+          var newCategories = this.categories ? this.categories.concat(childCategories) : childCategories.slice();
+          var childLogger = defaultLogger.getLogger.apply(defaultLogger,newCategories);
+          childLogger.childCategories = childCategories;
+          childLogger.newCategories =newCategories;
+          return childLogger;
```

Comment:
Similarly, I think this whole method just becomes `defaultLogger.getLogger(...keyHierarchy, ...arguments)`.